### DISCUSSION TITLE
dbmgr不能正常启动

### DISCUSSION THREAD

**Discussion by thoreaw** [kbengine/kbengine#1470] [Q&A]
[dbmgr.2021-07-13.log](https://github.com/kbengine/kbengine/files/6810480/dbmgr.2021-07-13.log)
[interfaces.2021-07-13.log](https://github.com/kbengine/kbengine/files/6810481/interfaces.2021-07-13.log)
[logger_baseapp.2021-07-13.log](https://github.com/kbengine/kbengine/files/6810482/logger_baseapp.2021-07-13.log)
[logger_baseappmgr.2021-07-13.log](https://github.com/kbengine/kbengine/files/6810483/logger_baseappmgr.2021-07-13.log)
[machine.log](https://github.com/kbengine/kbengine/files/6810484/machine.log)

  我使用的是阿里云的服务器linux版本CentOSv7.2日志见附件，dbmgr启动到一半就没有再启动了没有明显的报错
  INFO root [2021-07-13 23:41:56,312] - -----------------------------------------------------------------------------------------

  INFO root [2021-07-13 23:41:56,312] - Resmgr::initialize: KBE_ROOT=/root/kbengine/
  INFO root [2021-07-13 23:41:56,312] - Resmgr::initialize: KBE_RES_PATH=/root/kbengine/kbe/res/:/root/kbengine/server_assets/:/root/kbengine/server_assets/res/:/root/kbengine/server_assets/scripts/
  INFO root [2021-07-13 23:41:56,312] - Resmgr::initialize: KBE_BIN_PATH=/root/kbengine/kbe/bin/server/
  INFO root [2021-07-13 23:41:56,312] - NetworkInterface::initialize(INTERNAL-TCP): Creating on interface '172.30.170.247' (= 172.30.170.247)
  WARN root [2021-07-13 23:41:56,313] - NetworkInterface::initialize(INTERNAL-TCP): Operating with a receive buffer of only 425984 bytes (instead of 16777216)
  WARN root [2021-07-13 23:41:56,313] - NetworkInterface::initialize(INTERNAL-TCP): Operating with a send buffer of only 425984 bytes (instead of 16777216)
  INFO root [2021-07-13 23:41:56,313] - NetworkInterface::initialize(INTERNAL-TCP): address 172.30.170.247:47913, SOMAXCONN=511.
  INFO root [2021-07-13 23:41:56,313] - server-configs:
  INFO root [2021-07-13 23:41:56,313] - 	internalTcpAddr : 172.30.170.247:47913
  INFO root [2021-07-13 23:41:56,313] - 	componentID : 3129688875332859700
  INFO root [2021-07-13 23:41:57,815] - ---- dbmgr Version: 2.5.8. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: 11178A1C367680938C8E416B64844ACD. Config: Hybrid64 X64. Built: 20:38:06 Jul 13 2021. AppID: 3129688875332859700. UID: 10999. PID: 4103 ----
  INFO root [2021-07-13 23:41:57,815] - Server wwiiserver01: Intel(R) Xeon(R) Platinum 8369HC CPU @ 3.30GHz x4 with 15992300kB RAM
  INFO root [2021-07-13 23:41:57,815] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2021-07-13 23:41:57,815] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2021-07-13 23:41:57,915] - Script::install(): paths=/root/kbengine/server_assets/scripts/common:/root/kbengine/server_assets/scripts/data:/root/kbengine/server_assets/scripts/user_type:/root/kbengine/server_assets/scripts/server_common:/root/kbengine/server_assets/scripts/db:/root/kbengine/kbe/res/scripts:/root/kbengine/kbe/res/scripts/common:/root/kbengine/kbe/res/scripts/common/lib-dynload:/root/kbengine/kbe/res/scripts/common/DLLs:/root/kbengine/kbe/res/scripts/common/Lib:/root/kbengine/kbe/res/scripts/common/Lib/site-packages:/root/kbengine/kbe/res/scripts/common/Lib/dist-packages.
  INFO root [2021-07-13 23:41:57,939] - Script::install(): is successfully, Python=(3.7.3 (default, Jul 13 2021, 20:25:36) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)])!
  INFO root [2021-07-13 23:41:57,939] - Script::install(): KBExtra is successfully!

**Comment by huhubis** [comment]
看coredump。  core是不是没有打开

**Reply by thoreaw** [reply]
打开了的调试如下：
[wwii@wwiiserver01 server_assets]$ gdb /root/kbengine/kbe/bin/server/dbmgr core.3576
GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-120.el7
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /root/kbengine/kbe/bin/server/dbmgr...done.
[New LWP 3576]
[New LWP 3588]
[New LWP 3589]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `/root/kbengine/kbe/bin/server//dbmgr --cid=3129652375332859700 --gus=4'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007fdce9b90bce in SHA1_Init () from /usr/lib64/mysql/libmysqlclient.so.18
Missing separate debuginfos, use: debuginfo-install glibc-2.17-157.el7_3.5.x86_64 libgcc-4.8.5-44.el7.x86_64 libstdc++-4.8.5-44.el7.x86_64
(gdb) bt
#0  0x00007fdce9b90bce in SHA1_Init () from /usr/lib64/mysql/libmysqlclient.so.18
#1  0x00007fdce9b64d14 in init () from /usr/lib64/mysql/libmysqlclient.so.18
#2  0x0000000000adb52f in EVP_Digest ()
#3  0x00007fdce9c0243c in RSA_padding_add_PKCS1_OAEP_mgf1 () from /usr/lib64/mysql/libmysqlclient.so.18
#4  0x00007fdce9c02648 in RSA_padding_add_PKCS1_OAEP () from /usr/lib64/mysql/libmysqlclient.so.18
#5  0x00007fdce9b89114 in rsa_ossl_public_encrypt () from /usr/lib64/mysql/libmysqlclient.so.18
#6  0x00007fdce9b86af6 in RSA_public_encrypt () from /usr/lib64/mysql/libmysqlclient.so.18
#7  0x00000000007c50c5 in KBEngine::KBE_RSA::encrypt (this=0x7ffea60c92a0, instr="sky2005", outCertifdata="") at rsa.cpp:212
#8  0x00000000007c5205 in KBEngine::KBE_RSA::encrypt (this=<optimized out>, instr="sky2005") at rsa.cpp:195
#9  0x000000000078b4ac in KBEngine::DBUtil::initialize () at db_interface.cpp:93
#10 0x0000000000671908 in KBEngine::Dbmgr::initDB (this=this@entry=0x7ffea60c9620) at dbmgr.cpp:423
#11 0x0000000000672f88 in KBEngine::Dbmgr::initializeEnd (this=0x7ffea60c9620) at dbmgr.cpp:327
#12 0x000000000071b92d in KBEngine::ServerApp::initialize (this=0x7ffea60c9620) at serverapp.cpp:151
#13 0x00000000006b7b72 in KBEngine::kbeMainT<KBEngine::Dbmgr> (argc=<optimized out>, argv=<optimized out>, componentType=componentType@entry=KBEngine::DBMGR_TYPE, 
    extlisteningTcpPort_min=extlisteningTcpPort_min@entry=-1, extlisteningTcpPort_max=extlisteningTcpPort_max@entry=-1, 
    extlisteningUdpPort_min=extlisteningUdpPort_min@entry=-1, extlisteningUdpPort_max=extlisteningUdpPort_max@entry=-1, 
    extlisteningInterface=extlisteningInterface@entry=0xc283a2 "", intlisteningPort_min=intlisteningPort_min@entry=0, 
    intlisteningPort_max=intlisteningPort_max@entry=0, intlisteningInterface=0x10a9972 <KBEngine::g_serverConfig+5106> "")
    at /root/kbengine/kbe/src/lib/server/kbemain.h:204
#14 0x00000000006940b7 in kbeMain (argc=<optimized out>, argv=<optimized out>) at main.cpp:61
#15 0x00007fdce8be8b35 in __libc_start_main () from /lib64/libc.so.6
#16 0x000000000065b354 in _start ()

**Reply by thoreaw** [reply]
根据提示还安装了Debuginfo、glibc-2.17-157.el7_3.5.x86_64、libgcc-4.8.5-44.el7.x86_64因为之前阿里云里是没有的，后面调试如下：
[wwii@wwiiserver01 server_assets]$ gdb /root/kbengine/kbe/bin/server/dbmgr core.5125
GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-120.el7
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /root/kbengine/kbe/bin/server/dbmgr...done.
[New LWP 5125]
[New LWP 5139]
[New LWP 5140]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `/root/kbengine/kbe/bin/server//dbmgr --cid=3129652375332859700 --gus=4'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007fd0da2edbce in SHA1_Init () from /usr/lib64/mysql/libmysqlclient.so.18
(gdb) bt
#0  0x00007fd0da2edbce in SHA1_Init () from /usr/lib64/mysql/libmysqlclient.so.18
#1  0x00007fd0da2c1d14 in init () from /usr/lib64/mysql/libmysqlclient.so.18
#2  0x0000000000adb52f in EVP_Digest ()
#3  0x00007fd0da35f43c in RSA_padding_add_PKCS1_OAEP_mgf1 () from /usr/lib64/mysql/libmysqlclient.so.18
#4  0x00007fd0da35f648 in RSA_padding_add_PKCS1_OAEP () from /usr/lib64/mysql/libmysqlclient.so.18
#5  0x00007fd0da2e6114 in rsa_ossl_public_encrypt () from /usr/lib64/mysql/libmysqlclient.so.18
#6  0x00007fd0da2e3af6 in RSA_public_encrypt () from /usr/lib64/mysql/libmysqlclient.so.18
#7  0x00000000007c50c5 in KBEngine::KBE_RSA::encrypt (this=0x7ffdb29d0260, instr="sky2005", outCertifdata="") at rsa.cpp:212
#8  0x00000000007c5205 in KBEngine::KBE_RSA::encrypt (this=<optimized out>, instr="sky2005") at rsa.cpp:195
#9  0x000000000078b4ac in KBEngine::DBUtil::initialize () at db_interface.cpp:93
#10 0x0000000000671908 in KBEngine::Dbmgr::initDB (this=this@entry=0x7ffdb29d05e0) at dbmgr.cpp:423
#11 0x0000000000672f88 in KBEngine::Dbmgr::initializeEnd (this=0x7ffdb29d05e0) at dbmgr.cpp:327
#12 0x000000000071b92d in KBEngine::ServerApp::initialize (this=0x7ffdb29d05e0) at serverapp.cpp:151
#13 0x00000000006b7b72 in KBEngine::kbeMainT<KBEngine::Dbmgr> (argc=<optimized out>, argv=<optimized out>, 
    componentType=componentType@entry=KBEngine::DBMGR_TYPE, extlisteningTcpPort_min=extlisteningTcpPort_min@entry=-1, 
    extlisteningTcpPort_max=extlisteningTcpPort_max@entry=-1, extlisteningUdpPort_min=extlisteningUdpPort_min@entry=-1, 
    extlisteningUdpPort_max=extlisteningUdpPort_max@entry=-1, extlisteningInterface=extlisteningInterface@entry=0xc283a2 "", 
    intlisteningPort_min=intlisteningPort_min@entry=0, intlisteningPort_max=intlisteningPort_max@entry=0, 
    intlisteningInterface=0x10a9972 <KBEngine::g_serverConfig+5106> "") at /root/kbengine/kbe/src/lib/server/kbemain.h:204
#14 0x00000000006940b7 in kbeMain (argc=<optimized out>, argv=<optimized out>) at main.cpp:61
#15 0x00007fd0d9345b35 in __libc_start_main (main=0x64c5e0 <main(int, char**)>, argc=3, ubp_av=0x7ffdb29d0a98, init=<optimized out>, fini=<optimized out>, 
    rtld_fini=<optimized out>, stack_end=0x7ffdb29d0a88) at ../csu/libc-start.c:274
#16 0x000000000065b354 in _start ()

**Reply by huhubis** [reply]
https://github.com/kbengine/kbengine/issues/1270

**Comment by huhubis** [comment]
https://github.com/kbengine/kbengine/issues/1270

**Comment by thoreaw** [comment]
解决方法1:
修改common.mak: 
重新编译后报错通过不了
解决方法2：
安装适合版本的ssl库， 删除系统内高版本的
apt install --allow-downgrades -y
libssl1.0-dev
libmysqlclient20=5.7.21-1ubuntu1
libmysqlclient-dev=5.7.21-1ubuntu1
我使用的是CentOS7.2谁知道怎么删除才干净
